controller/adr.rst

```diff
@@ -0,0 +1,142 @@
+.. index::
+    single: Action Domain Responder approach
+
+How to implement the ADR pattern
+================================
+
+In Symfony, you're used to implement the MVC pattern and extending the default :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller`
+class.
+Since the 3.3 update, Symfony is capable of using natively the ADR approach.
+
+Updating your configuration
+---------------------------
+
+As the framework evolve, you must update the services.yml file in order to
+use the latest features of the DependencyInjection component, this way, here's the updates::
+
+    # ...
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        # Allow to load every actions
+        AppBundle\Action\:
+            resource: '../../src/AppBundle/Action/'
+            public: true
+
+Once the file is updated, delete your Controller folder and create an Action class using the ADR principles, i.e::
+
+    <?php
+
+    namespace AppBundle\Action;
+
+    use Twig\Environment;
+    use Symfony\Component\HttpFoundation\Response;
+
+    final class HelloAction
+    {
+        private $twig;
+
+        public function __construct(Environment $twig)
+        {
+            $this->twig = $twig;
+        }
```

Comment:
Thinking again about this approach, if we have just one method/controller, shouldn't be simpler inject the required dependencies into `__invoke()` directly? Thus, we are avoiding define one property per service argument, if so the action service need to be tagged with `controller.service_arguments`.

**After**
```php
final class HelloAction
{
    public function __invoke(Environment $twig): Response
    {
         return new Response($twig->render('default/index.html.twig'));
    }
}
```